File: mpp-firma-transport-c#/FirmaNetworking/Utils/RPCConcurrentServer.cs

using FirmaNetworking.Utils;
using System;
using System.Net.Sockets;
using System.Threading;
using FirmaService;
using FirmaNetworking;

namespace FirmaNetworking.Utils
{
    public class RPCConcurrentServer : AbstractConcurrentServer
    {
        private IFirmaServices server;
        private FirmaClientRPCWorker worker;
        public RPCConcurrentServer(string host, int port, IFirmaServices server) : base(host, port)
        {
            this.server = server;
            Console.WriteLine("RPCConcurrentServer...");
        }
        protected override Thread CreateWorker(TcpClient client)
        {
            worker = new FirmaClientRPCWorker(server, client);
            return new Thread(new ThreadStart(worker.Run));
        }

    }
}

File: MPP-firma-transport-java/FirmaNetworking/src/main/java/org/example/Utils/RPCConcurrentServer.java

package org.example.Utils;

import org.example.IFirmaServices;
import org.example.FirmaClientRPCWorker;

import java.net.Socket;

public class RPCConcurrentServer extends AbstractConcurrentServer {

    private IFirmaServices firmaServices;
    public RPCConcurrentServer(int port, IFirmaServices firmaServices) {
        super(port);
        this.firmaServices = firmaServices;
        System.out.println("RPCConcurrentServer");
    }

    @Override
    protected Thread createWorker(Socket client) {
        FirmaClientRPCWorker worker = new FirmaClientRPCWorker(firmaServices, client);
        Thread tw = new Thread(worker);
        return tw;
    }
}
